### PR TITLE
Add Spring Boot-based User Management API with DTOs, JPA, and Full Test Coverage

### PR DESCRIPTION
**Pull request by AlexeyBudkov**
### Changes
- Added Spring Boot modules: Web, Data JPA, Validation.
- Implemented CRUD endpoints for users:
  - Create, Read (by ID and email), Update, Delete.
- Introduced DTOs (`UserRequest`, `UserResponse`) to avoid exposing entity classes.
- Replaced manual Hibernate DAO with Spring Data JPA repository.
- Added `GlobalExceptionHandler` for unified JSON error responses.
- Wrote comprehensive tests:
  - Unit tests for service layer.
  - Integration tests with H2.
  - Controller tests (happy-path and edge cases) using MockMvc.
  - Direct tests for `GlobalExceptionHandler`.
- Prepared Postman collection for API testing.

### How to Test
Copy the API into a JSON file and add the file to Postman.

{
  "info": {
    "name": "User Service API",
    "_postman_id": "b8d8f1a0-1234-5678-90ab-cdef12345678",
    "description": "Коллекция для тестирования CRUD API user-service",
    "schema": "https://schema.getpostman.com/json/collection/v2.1.0/collection.json"
  },
  "item": [
    {
      "name": "Create User",
      "request": {
        "method": "POST",
        "header": [
          { "key": "Content-Type", "value": "application/json" }
        ],
        "body": {
          "mode": "raw",
          "raw": "{\n  \"name\": \"Alex\",\n  \"email\": \"alex@example.com\",\n  \"age\": 30\n}"
        },
        "url": { "raw": "http://localhost:8080/api/users", "protocol": "http", "host": ["localhost"], "port": "8080", "path": ["api", "users"] }
      }
    },
    {
      "name": "Get User by ID",
      "request": {
        "method": "GET",
        "url": { "raw": "http://localhost:8080/api/users/1", "protocol": "http", "host": ["localhost"], "port": "8080", "path": ["api", "users", "1"] }
      }
    },
    {
      "name": "Get User by Email",
      "request": {
        "method": "GET",
        "url": {
          "raw": "http://localhost:8080/api/users/by-email?email=alex@example.com",
          "protocol": "http",
          "host": ["localhost"],
          "port": "8080",
          "path": ["api", "users", "by-email"],
          "query": [{ "key": "email", "value": "alex@example.com" }]
        }
      }
    },
    {
      "name": "Get All Users",
      "request": {
        "method": "GET",
        "url": { "raw": "http://localhost:8080/api/users", "protocol": "http", "host": ["localhost"], "port": "8080", "path": ["api", "users"] }
      }
    },
    {
      "name": "Update User",
      "request": {
        "method": "PUT",
        "header": [
          { "key": "Content-Type", "value": "application/json" }
        ],
        "body": {
          "mode": "raw",
          "raw": "{\n  \"name\": \"Alex Updated\",\n  \"email\": \"alex@example.com\",\n  \"age\": 31\n}"
        },
        "url": { "raw": "http://localhost:8080/api/users/1", "protocol": "http", "host": ["localhost"], "port": "8080", "path": ["api", "users", "1"] }
      }
    },
    {
      "name": "Delete User",
      "request": {
        "method": "DELETE",
        "url": { "raw": "http://localhost:8080/api/users/1", "protocol": "http", "host": ["localhost"], "port": "8080", "path": ["api", "users", "1"] }
      }
    },
    {
      "name": "Create User - Invalid Name",
      "request": {
        "method": "POST",
        "header": [
          { "key": "Content-Type", "value": "application/json" }
        ],
        "body": {
          "mode": "raw",
          "raw": "{\n  \"name\": \"A\",\n  \"email\": \"short@example.com\",\n  \"age\": 25\n}"
        },
        "url": { "raw": "http://localhost:8080/api/users", "protocol": "http", "host": ["localhost"], "port": "8080", "path": ["api", "users"] }
      }
    },
    {
      "name": "Create User - Invalid Email",
      "request": {
        "method": "POST",
        "header": [
          { "key": "Content-Type", "value": "application/json" }
        ],
        "body": {
          "mode": "raw",
          "raw": "{\n  \"name\": \"Alex\",\n  \"email\": \"invalid-email\",\n  \"age\": 25\n}"
        },
        "url": { "raw": "http://localhost:8080/api/users", "protocol": "http", "host": ["localhost"], "port": "8080", "path": ["api", "users"] }
      }
    },
    {
      "name": "Get Non-existing User",
      "request": {
        "method": "GET",
        "url": { "raw": "http://localhost:8080/api/users/999", "protocol": "http", "host": ["localhost"], "port": "8080", "path": ["api", "users", "999"] }
      }
    }
  ]
}

### CODE REVIEW

File: src/main/java/com/example/userservice/controller/UserController.java [hunk 1]
```diff
@@ -1,36 +1,52 @@
...
-        this.userService = new UserServiceImpl(userDao);
+    public UserController(UserService service) {
+        this.service = service;
     }
 
-    public Optional<User> getUser(Long id) {
-        return userService.getUserById(id);
+    @PostMapping
+    @ResponseStatus(HttpStatus.CREATED)
+    public UserResponse create(@RequestBody UserRequest request) {
```
Comment (IntensiveJava) [inline src/main/java/com/example/userservice/controller/UserController.java:23]: рекомендую валидацию

File: src/main/java/com/example/userservice/dto/UserResponse.java [hunk 2]
```diff
@@ -0,0 +1,85 @@
+package com.example.userservice.dto;
+
+import java.time.Instant;
+
+public class UserResponse {
```
Comment (IntensiveJava) [inline src/main/java/com/example/userservice/dto/UserResponse.java:5]: можно record class

File: src/main/java/com/example/userservice/dto/UserResponse.java [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+
+    private Long id;
+    private String name;
+    private String email;
+    private Integer age;
+    private Instant createdAt;
+    private Instant updatedAt;
+
+    public UserResponse() {
```
Comment (IntensiveJava) [inline src/main/java/com/example/userservice/dto/UserResponse.java:14]: Ломбок?

File: src/main/java/com/example/userservice/mapper/UserMapper.java [hunk 4]
```diff
@@ -0,0 +1,35 @@
+package com.example.userservice.mapper;
+
+import com.example.userservice.dto.UserRequest;
+import com.example.userservice.dto.UserResponse;
+import com.example.userservice.entity.User;
+
+public class UserMapper {
+
+    public static User toEntity(UserRequest req) {
```
Comment (IntensiveJava) [inline src/main/java/com/example/userservice/mapper/UserMapper.java:9]: Опять req)))
почему не request?

File: src/main/java/com/example/userservice/service/impl/UserServiceImpl.java [hunk 5]
```diff
@@ -0,0 +1,142 @@
...
+import org.slf4j.LoggerFactory;
+import org.springframework.dao.DataIntegrityViolationException;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+
+@Service
+public class UserServiceImpl implements UserService {
+
+    private static final Logger log = LoggerFactory.getLogger(UserServiceImpl.class);
```
Comment (IntensiveJava) [inline src/main/java/com/example/userservice/service/impl/UserServiceImpl.java:20]: есть аннотация для логгера

File: src/main/java/com/example/userservice/service/impl/UserServiceImpl.java [hunk 6]
```diff
@@ -0,0 +1,142 @@
...
+
+    private final UserRepository repo;
+
+    public UserServiceImpl(UserRepository repo) {
+        this.repo = repo;
+    }
+
+    @Override
+    public UserResponse create(UserRequest request) {
+        validateUser(request);
```
Comment (IntensiveJava) [inline src/main/java/com/example/userservice/service/impl/UserServiceImpl.java:30]: валидация запроса на уровне контроллера

File: src/main/java/com/example/userservice/service/impl/UserServiceImpl.java [hunk 7]
```diff
@@ -0,0 +1,142 @@
...
+        if (users.isEmpty()) {
+            log.warn("Список пользователей пуст");
+        } else {
+            log.info("Найдено {} пользователей", users.size());
+        }
+        return users.stream().map(UserMapper::toResponse).toList();
+    }
+
+    @Override
+    public UserResponse update(Long id, UserRequest request) {
```
Comment (IntensiveJava) [inline src/main/java/com/example/userservice/service/impl/UserServiceImpl.java:73]: может айдишник в запрос поместить?

File: src/test/java/com/example/userservice/service/UserServiceImplTest.java [hunk 8]
```diff
@@ -1,110 +1,78 @@
...
-    @Test
-    void createUser_InvalidEmail_ThrowsException() {
-        testUser.setEmail("invalid-email");
-
-        IllegalArgumentException ex = assertThrows(IllegalArgumentException.class,
-                () -> userService.createUser(testUser));
-
-        assertEquals("Некорректный email", ex.getMessage());
-        verify(userDao, never()).create(any());
+        userRepository = mock(UserRepository.class);
```
Comment (IntensiveJava) [inline src/test/java/com/example/userservice/service/UserServiceImplTest.java:28]: почему не Аннотации?
